src/main/java/io/aiven/connect/elasticsearch/clientwrapper/AivenElasticsearchClientWrapper.java

```diff
@@ -0,0 +1,376 @@
+            this.version = getServerVersion();
+        } catch (final IOException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to connection error:",
+                e
+            );
+        }
+    }
+
+    // visible for testing
+    public AivenElasticsearchClientWrapper(final String address) {
+        try {
+            final Map<String, String> props = new HashMap<>();
+            props.put(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG, address);
+            props.put(ElasticsearchSinkConnectorConfig.TYPE_NAME_CONFIG, "kafka-connect");
+            this.elasticClient = getElasticsearchClient(new ElasticsearchSinkConnectorConfig(props));
+            this.version = getServerVersion();
+        } catch (final IOException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to connection error:",
+                e
+            );
+        } catch (final ConfigException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to configuration error:",
+                e
+            );
+        }
+    }
+
+    // visible for testing
+    public AivenElasticsearchClientWrapper(final ElasticsearchSinkConnectorConfig config) {
+        try {
+            this.elasticClient = getElasticsearchClient(config);
+            this.version = getServerVersion();
+        } catch (final IOException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to connection error:",
+                e
+            );
+        } catch (final ConfigException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to configuration error:",
+                e
+            );
+        }
+    }
+
+    private Version getServerVersion() throws IOException {
+        final String esVersion = this.elasticClient.info(RequestOptions.DEFAULT).getVersion().getNumber();
+        return matchVersionString(esVersion);
+    }
+
+    private Version matchVersionString(final String esVersion) {
+        // Default to newest version for forward compatibility
+        final Version defaultVersion = Version.ES_V8;
+        if (esVersion == null) {
+            LOG.warn("Couldn't get Elasticsearch version, version is null");
+            return defaultVersion;
+        } else if (esVersion.startsWith("1.")) {
+            return Version.ES_V1;
+        } else if (esVersion.startsWith("2.")) {
+            return Version.ES_V2;
+        } else if (esVersion.startsWith("5.")) {
+            return Version.ES_V5;
+        } else if (esVersion.startsWith("6.")) {
+            return Version.ES_V6;
+        } else if (esVersion.startsWith("7.")) {
+            return Version.ES_V7;
+        } else if (esVersion.startsWith("8.")) {
+            return Version.ES_V8;
+        }
+        return defaultVersion;
+    }
```

Comment:
are we sure we want to default something?
If we cannot read the version isn't something that should pop up instead of assigning a default?

Maybe we can add an option to specify this behaviour but if you cannot parse the version of the protocol maybe its a good moment of popping up an error (like if a new incompatibile version arrives we should manually check and add the fact that we are supporting that version with this version of the connector)

Reply:
Changing this based on the later comment and removing the unsupported versions (1, 2, 3, 4, 5, 6).